### PR TITLE
[TECH-527]Book Command - Dispatcher & Handler

### PR DESCRIPTION
**Pull request by LubaibaZ**

### CODE REVIEW

File: book.cmd/src/main/java/com/happystays/book/cmd/api/commands/PaymentInfo.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import lombok.Data;
+
+@Data
+public class PaymentInfo {
+    private String cardCode;
+    private String cardNumber;
+    private String seriescode;
+    private String expiryDate;
+    private String cardHolderName;
```
Comment (JeanMariaJose) [inline book.cmd/src/main/java/com/happystays/book/cmd/api/commands/PaymentInfo.java:12]: Do we need a paymentType? can we ename cardcode as cardType and seriesCode as cvv

File: book.cmd/src/main/java/com/happystays/book/cmd/api/commands/CancelInfo.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.happystays.book.cmd.api.commands;
+
+import lombok.Data;
+
+@Data
+public class CancelInfo {
+    private boolean cancellable;
```
Comment (JeanMariaJose) [inline book.cmd/src/main/java/com/happystays/book/cmd/api/commands/CancelInfo.java]: can we rename this as isCancellable?

File: book.cmd/src/main/java/com/happystays/book/cmd/api/commands/BookCommand.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    private String ratePlanId;
+    private int guestcount;
+    private int guestAge;
+    private String startDate;
+    private String endDate;
+    private PaymentInfo paymentInfo;
+    private String hotelCode;
+    private CustomerInfo customerInfo;
+    private CancelInfo cancelInfo;
+    private String guaranteeIndicator;
```
Comment (JeanMariaJose) [inline book.cmd/src/main/java/com/happystays/book/cmd/api/commands/BookCommand.java]: what is this guaranteeIndicator? we wont get this UI. I this we may need to hardcode this value in DB as we wont get this either from UI or in Book RS

File: book.cmd/src/main/java/com/happystays/book/cmd/api/commands/BookCommand.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+public class BookCommand extends BaseCommand {
+    private int numberOfRooms;
+    private String ratePlanId;
+    private int guestcount;
+    private int guestAge;
+    private String startDate;
+    private String endDate;
+    private PaymentInfo paymentInfo;
+    private String hotelCode;
+    private CustomerInfo customerInfo;
```
Comment (JeanMariaJose) [inline book.cmd/src/main/java/com/happystays/book/cmd/api/commands/BookCommand.java]: Can we rename as UserInfo

File: book.cmd/src/main/java/com/happystays/book/cmd/api/commands/BookCommand.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.happystay.cqrs.core.commands.BaseCommand;
+import lombok.Data;
+
+@Data
+public class BookCommand extends BaseCommand {
+    private int numberOfRooms;
+    private String ratePlanId;
+    private int guestcount;
+    private int guestAge;
```
Comment (JeanMariaJose) [inline book.cmd/src/main/java/com/happystays/book/cmd/api/commands/BookCommand.java]: we wont get age from UI..I think this is not required for now

File: book.cmd/src/main/java/com/happystays/book/cmd/api/commands/BookCommand.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    private String startDate;
+    private String endDate;
+    private PaymentInfo paymentInfo;
+    private String hotelCode;
+    private CustomerInfo customerInfo;
+    private CancelInfo cancelInfo;
```
Comment (JeanMariaJose) [inline book.cmd/src/main/java/com/happystays/book/cmd/api/commands/BookCommand.java]: can we rename as CancellationInfo

File: book.cmd/src/main/java/com/happystays/book/cmd/api/commands/CancelInfo.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    private String cancelDate;
```
Comment (JeanMariaJose) [inline book.cmd/src/main/java/com/happystays/book/cmd/api/commands/CancelInfo.java]: can we rename this as cancellationDate?

File: book.cmd/src/main/java/com/happystays/book/cmd/api/commands/CancelInfo.java [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+    private String cancelPolicy;
```
Comment (JeanMariaJose) [inline book.cmd/src/main/java/com/happystays/book/cmd/api/commands/CancelInfo.java]: rename as cancellationPolicy

File: book.cmd/src/main/java/com/happystays/book/cmd/api/commands/PaymentInfo.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package com.happystays.book.cmd.api.commands;
+
+import lombok.Data;
+
+@Data
```
Comment (JeanMariaJose) [inline book.cmd/src/main/java/com/happystays/book/cmd/api/commands/PaymentInfo.java:5]: Do we need @NoArgsConstructor here?

**Review by JeanMariaJose — CHANGES_REQUESTED** [review]

File: book.cmd/src/main/java/com/happystays/book/cmd/api/commands/PaymentInfo.java (hunk 9 above)
Comment (LubaibaZ) [reply book.cmd/src/main/java/com/happystays/book/cmd/api/commands/PaymentInfo.java:5]: I think only @Data is enough

File: book.cmd/src/main/java/com/happystays/book/cmd/api/commands/BookCommand.java (hunk 3 above)
Comment (LubaibaZ) [reply book.cmd/src/main/java/com/happystays/book/cmd/api/commands/BookCommand.java]: Will discuss and remove if not needed

Comment (LubaibaZ) [reply book.cmd/src/main/java/com/happystays/book/cmd/api/commands/BookCommand.java]: Removed the field

**Review by JeanMariaJose — APPROVED** [review]

**Pull request merged**